src/components/table.js

```diff
@@ -432,40 +480,76 @@ export const GridTable = forwardRefWithName('GridTable', ({
           'aria-label': `${ariaLabel} header row`, // The whole table is a tab stop so it needs a label
           'aria-readonly': null, // Clear out ARIA properties which have been moved one level up
-          width: width - scrollbarSize,
+          // Leave space for fixed columns.
+          width: width - scrollbarSize - totalFixedColumnsWidth,
           height: headerHeight,
-          columnWidth: ({ index }) => columns[index].width,
+          columnWidth: ({ index }) => unfixedColumns[index].width,
           rowHeight: headerHeight,
           rowCount: 1,
-          columnCount: columns.length,
-          cellRenderer: data => {
-            const field = columns[data.columnIndex].field
-            return div({
-              key: data.key,
-              role: 'columnheader',
-              // ARIA row and column indexes start with 1 rather than 0 https://www.digitala11y.com/aria-colindexproperties/
-              'aria-rowindex': 1, // The header row is 1
-              'aria-colindex': data.columnIndex + 1, // The first column is 1
-              'aria-sort': ariaSort(sort, field),
-              className: 'table-cell',
-              style: {
-                ...data.style,
-                ...styles.header(data.columnIndex, columns.length, { border }),
-                ...styleHeader(data)
-              }
-            }, [
-              columns[data.columnIndex].headerRenderer(data)
-            ])
+          columnCount: unfixedColumns.length,
+          cellRenderer: ({ columnIndex, ...cell }) => {
+            // columnIndex here is an index in the unfixedColumns array.
+            // Offset it to pass an index in the original columns array to renderHeaderCell.
+            return renderHeaderCell({ ...cell, columnIndex: columnIndex + numFixedColumns })
+          },
+          cellRangeRenderer: data => {
+            // The default renderer returns a flat array of all of the cells to render in the DOM
+            const cells = defaultCellRangeRenderer(data)
+
+            const {
+              horizontalOffsetAdjustment,
+              rowSizeAndPositionManager,
+              verticalOffsetAdjustment
+            } = data
+
+            const rowDatum = rowSizeAndPositionManager.getSizeAndPositionOfCell(0)
+
+            return [
+              ..._.map(columnIndex => {
+                return renderHeaderCell({
+                  key: `fixed-${columnIndex}`,
+                  columnIndex,
+                  rowIndex: 0,
+                  style: {
+                    height: rowDatum.size,
+                    left: fixedColumnOffsets[columnIndex] + horizontalOffsetAdjustment,
+                    position: 'fixed',
+                    top: rowDatum.offset + verticalOffsetAdjustment,
+                    width: fixedColumns[columnIndex].width,
+                    // Show header cell above body cells in fixed columns when vertically scrolling the grid.
+                    zIndex: 2
```

Comment:
Nit: below at line 616 use used 1 for zIndex. Can you make both values the same for consistency?

Reply:
Got it, thanks!